notebook/conversation.ipynb

```diff
@@ -0,0 +1,187 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "28c4c4dc-7107-45e5-9ef4-f6506a0d55e2",
+   "metadata": {},
+   "source": [
+    "To install AgentScope, you need to have Python 3.9 or higher installed."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6dbd5990-ed46-4727-84b1-d4131c5a6245",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install AgentScope"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4ccebee3-9a27-47d9-9133-40126b888593",
+   "metadata": {},
+   "source": [
+    "First, set the configs of the models you use."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c60227f7-7380-4f72-b653-bc7394cfaace",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import agentscope\n",
+    "from agentscope.agents import DialogAgent\n",
+    "from agentscope.agents.user_agent import UserAgent\n",
+    "from agentscope.pipelines.functional import sequentialpipeline\n",
+    "\n",
+    "agentscope.init(\n",
+    "    model_configs=[\n",
+    "        {\n",
+    "            \"type\": \"openai\",\n",
+    "            \"name\": \"gpt-3.5-turbo\",\n",
+    "            \"api_key\": \"xxx\",  # Load from env if not provided\n",
+    "            \"organization\": \"xxx\",  # Load from env if not provided\n",
+    "            \"generate_args\": {\n",
+    "                \"temperature\": 0.5,\n",
+    "            },\n",
+    "        },\n",
+    "        {\n",
+    "            \"type\": \"post_api\",\n",
+    "            \"name\": \"my_post_api\",\n",
+    "            \"api_url\": \"https://xxx\",\n",
+    "            \"headers\": {},\n",
+    "        },\n",
+    "    ],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "503f7e7a-7a93-468d-b88b-37540f607083",
+   "metadata": {},
+   "source": [
+    "Then, initialize two agents, one is used as an assistant agent, and the other one is a user agent."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "848270c3-b73b-482d-a426-49e0ac67dc39",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dialog_agent = DialogAgent(\n",
+    "    name=\"Assistant\",\n",
+    "    sys_prompt=\"You're a helpful assistant.\",\n",
+    "    model=\"gpt-3.5-turbo\",  # replace by your model config name\n",
+    ")\n",
+    "user_agent = UserAgent()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "128356259909067b",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "Start the conversation between user and assistant. Input \"exit\" to quit."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6057d6c5-a332-4949-84a7-6fbeda021561",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "x = None\n",
+    "while x is None or x.content != \"exit\":\n",
```

Comment:
Pls provide users a learning process here with two versions. The first version is without pipeline, and the second is with pipeline.

Reply:
E.g. to code a conversation easier, you can use pipeline in `agentscope.pipelines` and `agentscope.pipelines.functional`